Add bold and italic format guidance (#384)
grafana/writers-toolkit: docs/sources/write/style-guide/style-conventions/index.md

The output is the same in both cases.
However, the first is easier to review and edit and is less subject to the screen and text editor settings of each contributor.

## Text formatting

It's a good idea to take a consistent approach to bold, italic, and other text formats.

### Bold

> Use bold formatting, \*\* in Markdown, for UI elements. Don't use bold to draw attention to a word or phrase.

Reviewer: ```suggestion
Use bold formatting (\*\* in Markdown), for UI elements. Don't use bold to draw attention to a word or phrase.
```

### Italic

> Use italics formatting, \_ in Markdown, to emphasize a specific word or phrase. This is particularly useful when defining a term for the first time.

Reviewer: ```suggestion
Use italics formatting (\_ in Markdown), to emphasize a specific word or phrase. This is particularly useful when defining a term for the first time.
```

> **Recommended:** The Prometheus data model is arranged into _metrics_ that consist of a _timestamp_ and a _sample_.

Reviewer 2: This suggestion avoids the confusion of using bold in a different way to the docs.

```suggestion
For example:

> The Prometheus data model is arranged into _metrics_ that consist of a _timestamp_ and a _sample_.
```

The output is the same in both cases.
However, the first is easier to review and edit and is less subject to the screen and text editor settings of each contributor.

## Text formatting

Follow a consistent approach to bold, italic, and other text formats.

### Bold

Use bold formatting, \*\* in Markdown, for UI elements. Don't use bold to draw attention to a word or phrase.

### Italic

Use italics formatting, \_ in Markdown, to emphasize a specific word or phrase. This is particularly useful when defining a term for the first time.

> **Recommended:** The Prometheus data model is arranged into _metrics_ that consist of a _timestamp_ and a _sample_.

Reviewer 2: Do we need to mention this use case for bold emphasis?

Author: no, shorter is better

Reviewer 2: I personally found it odd that we immediately used bold emphasis immediately after we said that we shouldn't use bold to draw attention to a word or phrase.
